Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```
@@ -67,56 +67,56 @@ public async Task<InvoiceModel> PurchaseAsync(IEnumerable<LineItemModel> lineIte
+                beveragePrice.BeverageSizeId = price.BeverageSizeId;
 
-                var result = await localDbContext.BeverageCost.AsNoTracking().Select(c => c).ToArrayAsync();
+                await localDbContext.SaveChangesAsync();
 
-                return result.Select(ToModel).ToArray();
+                return;
             }
 
-            var beverageSizeDictionary = localDbContext.BeverageCost.AsNoTracking()
-                .Where(x => x.Beverage.Name == price.Beverage.Name)
-                .Select(c => new { c.BeverageSizeId, c.BeverageSize.Name }).ToDictionary(b => b.BeverageSizeId, b => b.Name);
 
-            if(beverageSizeDictionary.ContainsKey(price.BeverageSizeId))
+            if(price.BeverageId == 0)
             {
+                await localDbContext.BeverageCost.AddAsync(ToEntity(price));
 
-                var record = await localDbContext.BeverageCost
-                    .SingleOrDefaultAsync(c => c.BeverageSizeId == price.BeverageSizeId && c.Beverage.Name == price.Beverage.Name);
+                localDbContext.SaveChanges();
 
-                record.Cost = price.Cost;
-
-                await localDbContext.SaveChangesAsync();
-                    
+                return;
             }
-            else
-            {
 
-                price.BeverageId = beverageDictionary.FirstOrDefault(x => x.Value == price.Beverage.Name).Key;
+            var isBeverageAvailable = await localDbContext.BeverageCost.AsNoTracking()
+                .Where(c => c.BeverageId == price.BeverageId)
+                .AnyAsync();
 
+            if(isBeverageAvailable)
+            {
                 price.Beverage = null;
-                    
-                await localDbContext.AddAsync(ToEntity(price));
+
+                await localDbContext.BeverageCost.AddAsync(ToEntity(price));
 
                 await localDbContext.SaveChangesAsync();
-               
-            }
 
+                return;
```

While it is not necessary now. . . It is worth never to swallow errors. So returning an error message/logging the fact that a none-existing Id was given could save you a lot of trouble.